### PR TITLE
Np 45130 cristin import moving picture to nva type

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/DescriptionExtractor.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+public interface DescriptionExtractor {
+
+     default String extractDescription(String... fields) {
+         var validFields = Arrays.stream(fields)
+             .filter(Objects::nonNull)
+             .collect(Collectors.toList());
+         return String.join(", ", validFields);
+
```
Comment (truhacevkir) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/DescriptionExtractor.java:16]: I think it is possible do it that way:
```
return Arrays.stream(fields)
             .filter(Objects::nonNull)
             .collect(joining(","));
```

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/MovingPictureExtractor.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            : MovingPictureSubtype.create(MovingPictureSubtypeEnum.FILM);
+
+    }
+
+
+    private boolean hasUnknownLenght(ArtisticProductionTimeUnit timeUnit, String duration) {
+        return isNull(timeUnit)
+            || !timeUnit.timeUnitIsInMinutes()
+            || isNull(duration);
+    }
```
Comment (truhacevkir) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/MovingPictureExtractor.java:36]: Assume these rules are correct :)

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinSecondaryCategory.java [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
 public enum CristinSecondaryCategory {
```
Comment (StigNorland) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinSecondaryCategory.java:12]: remove the comments?

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/ArtisticProductionTimeUnit.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+public class ArtisticProductionTimeUnit {
+
+
+    //The only time codes that exist in the dataset are minutes or weeks.
+    @JsonIgnore
+    public static final String MINUTE = "MINUTT";
+
+    @JsonIgnore
+    public static final int MAX_SHORT_FILM_LENGTH_IN_MINUTES = 40;
```
Comment (StigNorland) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/ArtisticProductionTimeUnit.java:34]: Is this used?

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/MovingPictureExtractor.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import no.unit.nva.model.instancetypes.artistic.film.MovingPictureSubtype;
+import no.unit.nva.model.instancetypes.artistic.film.MovingPictureSubtypeEnum;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+public interface MovingPictureExtractor {
+
+    String MINUTE = "MINUTT";
```
Comment (StigNorland) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/MovingPictureExtractor.java:12]: This is also decalred public in `ArtisticProductionTimeUnit `

**Review by StigNorland — CHANGES_REQUESTED** [review]
check comments and possible duplicate vars

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinSecondaryCategory.java (hunk 3 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinSecondaryCategory.java:12]: I would like to keep those for further mapping of other types of cristin-results.
It is a bit time consuming to figure out wich cristin category contains what objects.

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/ArtisticProductionTimeUnit.java (hunk 4 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/ArtisticProductionTimeUnit.java:34]: Good catch. Fixed

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/MovingPictureExtractor.java (hunk 5 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/MovingPictureExtractor.java:12]: good catch. Fixed

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 6]
```diff
@@ -212,6 +215,13 @@ public static CristinObject randomObject(String secondaryCategory) {
                 return randomTvOrMedia(category);
             case WRITTEN_INTERVIEW:
                 return randomWrittenInterview();
+//            case MUSICAL_PERFORMANCE:
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:218]: Remnants from when this task was ginourmous and included mapping for all these subtypes. 
Ended up splitting the task, out-comment these.
But the task containing the mapping of these are under selected for development, so I let them exist as comments.

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/DescriptionExtractor.java (hunk 1 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/DescriptionExtractor.java:16]: Fixed

**Review by StigNorland — APPROVED** [review]

**Pull request merged**